curriculum/challenges/english/25-front-end-development/workshop-fruit-search-app/67ed044dc76cf4a441618f2c.md

```diff
@@ -0,0 +1,123 @@
+---
+id: 67ed044dc76cf4a441618f2c
+title: Step 5
+challengeType: 0
+dashedName: step-5
+---
+
+# --description--
+
+Create a `query` state variable and its setter function, `setQuery`, using the `useState` Hook.
+
+# --hints--
+
+You should use the array destructuring syntax for the `query` and `setQuery` variables.
+
+```js
+assert.match(code, /(const|let)\s+\[\s*(query)\s*,\s*(setQuery)\s*\]\s*/);
+```
+
+You should use the `useState` Hook.
+
+```js
+assert.match(code, /useState\(/);
+```
+
+Your `useState` Hook should have an initial value of an empty string (`''`).
+
+```js
+assert.match(code, /useState\(\s*['"]{2}\s*\)/);
```

Comment:
it looks like the other labs and workshops are using this approach to testing

here is an example from the color picker lab

````
You should pass `#ffffff` as the initial value to your `useState` call.

```js
async () => {
    const abuseState = __helpers.spyOn(React, "useState");
    const script = [...document.querySelectorAll("script")].find((s) => s.dataset.src === "index.jsx").innerText;
    const exports = {};
    const _a = eval(script);
    const _b = await __helpers.prepTestComponent(exports.ColorPicker);
    assert.match(abuseState.calls[0], /#ffffff/i);
}
```
````